packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -224,10 +224,7 @@ function queueImageResizing({ file, args = {} }) {
     .update(JSON.stringify(sortedArgs))
     .digest(`hex`)
 
-  const argsDigestShort = argsDigest.substr(argsDigest.length - 5)
-
-  const imgSrc = `/${file.internal
-    .contentDigest}-${argsDigestShort}.${fileExtension}`
+  const imgSrc = `/${file.name}-${argsDigest}.${fileExtension}`
```

Comment:
Why stop using the short version of the digest? The filenames got a lot longer with this change due to adding back original name + lengthening the digest.